CMSgov/cmcs-eregulations
# EREGCSC-1176 -- Better timestamps

Resolves [EREGCSC-1176](https://jiraent.cms.gov/browse/EREGCSC-1176) -- Give informative datestamps on footer & ToC pages

**Description**

Updates copy around app to better communicate the two types of dates used to denote data freshness:
1. When the regulation text was synced from eCFR
   - ex: "Part 457 up to date from eCFR as of Feb 1, 2022"
2. When the regulation text itself was updated/amended
   - ex: "last amended Jan 27, 2022"

**This pull request changes...**

- Updates copy text on footer, subpart (reader) view, and Part Table of Contents (ToC) view to reflect date when text was synced from eCFR
- Updates copy text on subpart (reader) view and Part Table of Contents (ToC) view to also show latest "version" date, which is the date the regulation text itself was last updated/amended
- Formats dates per Figma comp (no leading zero on single-digit days)
- links directly to selected part in eCFR when on reader view and ToC veiw
- creates new `url_formatters` template tag file with utility function to create eCFR part link
- updates python views when needed to include needed information in django template
- updates Cypress test

**Steps to manually verify this change**

1. Visit [figma comp](https://www.figma.com/file/ALyNx9w9OX12yWUwjM7IID/eRegs-Datestamps?node-id=1507%3A45) to see annotated requirements
2. Visit [experimental deployment](https://hkc3k5a5n6.execute-api.us-east-1.amazonaws.com/dev499) and match footer text and date format with footer in figma comp
3. Visit the part ToC view (example: [Part 433](https://hkc3k5a5n6.execute-api.us-east-1.amazonaws.com/dev499/42/433/) ToC) and make sure copy text matches figma and the eCFR link goes to proper part in eCFR
4. Visit reader view (example: [Part 433 Subpart A](https://hkc3k5a5n6.execute-api.us-east-1.amazonaws.com/dev499/42/433/Subpart-A/2021-03-01/) reader view) and make sure copy text matches figma and eCFE link goes to proper part in eCFR
